addons/point_of_sale/models/pos_order.py

```diff
@@ -611,6 +612,187 @@ def _prepare_invoice_vals(self):
+                expense_account = stock_move.product_id._get_product_accounts()['expense']
+                stock_output_account = stock_move.product_id.categ_id.property_stock_account_output_categ_id
+                balance = -sum(stock_move.stock_valuation_layer_ids.mapped('value'))
+                aml_vals_list_per_nature['stock'].append({
+                    'name': _("Stock input for %s", stock_move.product_id.name),
+                    'account_id': expense_account.id,
+                    'partner_id': commercial_partner.id,
+                    'currency_id': self.company_id.currency_id.id,
+                    'amount_currency': balance,
+                    'balance': balance,
+                })
+                aml_vals_list_per_nature['stock'].append({
+                    'name': _("Stock output for %s", stock_move.product_id.name),
+                    'account_id': stock_output_account.id,
+                    'partner_id': commercial_partner.id,
+                    'currency_id': self.company_id.currency_id.id,
+                    'amount_currency': -balance,
+                    'balance': -balance,
+                })
+
+        # Payment terms.
+        pos_receivable_account = self.company_id.account_default_pos_receivable_account_id
+        aml_vals_list_per_nature['payment_terms'].append({
+            'name': f"{pos_receivable_account.code} {pos_receivable_account.code}",
+            'account_id': pos_receivable_account.id,
+            'partner_id': commercial_partner.id,
+            'currency_id': self.currency_id.id,
+            'amount_currency': -total_amount_currency,
+            'balance': -total_balance,
+        })
+
+        return aml_vals_list_per_nature
+
+    def _create_misc_reversal_move(self, payment_moves):
+        """ Create a misc move to reverse this POS order and "remove" it from the POS closing entry.
+        This is done by taking data from the order and using it to somewhat replicate the resulting entry in order to
+        reverse partially the movements done ine the POS closing entry.
+        """
+        aml_values_list_per_nature = self._prepare_aml_values_list_per_nature()
+        move_lines = []
+        for aml_values_list in aml_values_list_per_nature.values():
+            for aml_values in aml_values_list:
+                aml_values['balance'] = -aml_values['balance']
+                aml_values['amount_currency'] = -aml_values['amount_currency']
+                move_lines.append(aml_values)
+
+        # Make a move with all the lines.
+        reversal_entry = self.env['account.move'].with_context(default_journal_id=self.config_id.journal_id.id).create({
+            'journal_id': self.config_id.journal_id.id,
+            'date': fields.Date.today(),
+            'ref': _('Reversal of POS closing entry %s for %s') % (self.session_move_id.name, self.name),
+            'invoice_line_ids': [Command.create(aml_value) for aml_value in move_lines],
```

Comment:
Command is not used in this file, to keep consistency, you can remove the import of it and use "0"

Reply:
But isn't Command the new standard that we should be using? What about replacing the other ones by commands then instead?

Reply:
@pimodoo It's not too late to use them ;)

Reply:
we'll replace them all when we'll do a refactoring of this file, it is not done here and so we should respect the guidelines of a file before the guidelines of odoo.

Reply:
@pimodoo Wtf you are talking about. Just cleanup step by step the new lines but don't refactor the whole file in this file. It will noise the diff

Reply:
Do what you want, I don't really care. Maybe next time avoid trying to make pos tasks, and let the pos team do it. It'll avoid you to be disrespectful in your futile comments. 

Regarding the code cleaning you could do in accounting module, you probably should first review the code merged in those modules instead of the code merged in point of sale